main/src/cgeo/geocaching/MainActivity.java

```diff
@@ -652,4 +662,31 @@ public void updateSelectedBottomNavItemId() {
             actionBar.setDisplayHomeAsUpEnabled(true);
         }
     }
+
+    public void displayActionItem(final int layout, final @StringRes int info, final Runnable action) {
+        final RelativeLayout l = findViewById(layout);
+        if (l != null) {
+            l.setVisibility(View.VISIBLE);
+            updateHomeBadge(1);
+            ((TextView) l.findViewById(R.id.action_item_info)).setText(info);
+            l.findViewById(R.id.action_item_button).setOnClickListener(v -> {
+                action.run();
+                l.setVisibility(View.GONE);
+                updateHomeBadge(-1);
+            });
+        }
+    }
+
+    public void updateHomeBadge(final int delta) {
+        synchronized (notificationCounter) {
+            notificationCounter.set(notificationCounter.get() + delta);
+            final BottomNavigationView bottomNavigation = findViewById(R.id.activity_bottom_navigation);
```

Comment:
The parent class `NavigationBarView` should be used, as we don't have a bottom navigation but a navigation rail in landscape mode.

Furthermore if you want to show a badge, this code should be part of `AbstractBottomNavigationActivity` so that it is displayed always if bottom navigation is visible.

Reply:
Thanks for your remarks, had already been working on this (moving to `AbstractBottomNavigatinActivity` etc.), but am currently stuck with merging the badge with `onLoginIssue` - how many times is this called, and under what conditions?
I'll update this PR as a WIP to let you see the current state.